crates/subspace-solving/src/codec.rs

```diff
@@ -135,14 +135,20 @@ impl SubspaceCodec {
         if self.cuda_available {
             let pieces_to_process = pieces.len() / PIECE_SIZE / GPU_PIECE_BLOCK * GPU_PIECE_BLOCK;
 
-            self.batch_encode_cuda(
+            let cuda_result = self.batch_encode_cuda(
                 &mut pieces[..pieces_to_process * PIECE_SIZE],
                 &piece_indexes[..pieces_to_process],
-            )?;
-
-            // Leave the rest for CPU
-            pieces = &mut pieces[pieces_to_process * PIECE_SIZE..];
-            piece_indexes = &piece_indexes[pieces_to_process..];
+            );
+
+            // check if CUDA operation was successful
+            if let Err(e) = cuda_result {
```

Comment:
So this check will leave data dedicated to GPU completely unencoded?

Reply:
No. 
When CUDA is not available, we are performing `self.batch_encode_cpu(pieces, piece_indexes)?;`
And when CUDA is working, we are updating 
```
pieces = &mut pieces[pieces_to_process * PIECE_SIZE..];
piece_indexes = &piece_indexes[pieces_to_process..];
```

So, if we do not update these when CUDA creates an error, basically CPU will handle all the pieces